packages/@aws-cdk/aws-servicecatalog/lib/product-stack-asset-bucket.ts

```diff
@@ -0,0 +1,114 @@
+import { IBucket, BlockPublicAccess, Bucket, BucketEncryption } from '@aws-cdk/aws-s3';
+import { BucketDeployment, ISource, Source } from '@aws-cdk/aws-s3-deployment';
+import * as cdk from '@aws-cdk/core';
+import { Construct, IConstruct } from 'constructs';
+import { hashValues } from './private/util';
+/**
+ * Product stack asset bucket props.
+ */
+export interface ProductStackAssetBucketProps {
+  /**
+   * Name of s3 asset bucket deployed
+   *
+   * @default - None
+   */
+  readonly assetBucketName?: string;
+}
+
+/**
+ * A Service Catalog product stack asset bucket, which is similar in form to an Amazon S3 bucket.
+ * You can store multiple product stack assets and collectively deploy them to S3.
+ */
+export class ProductStackAssetBucket extends Construct {
+  private readonly bucketName: string;
+  private readonly bucket: IBucket;
+  private readonly assets: ISource[];
+
+  constructor(scope: Construct, id: string, props: ProductStackAssetBucketProps = {}) {
+    super(scope, id);
+
+    if (props.assetBucketName) {
+      this.bucketName = props.assetBucketName;
+    } else {
+      this.bucketName = this.generateBucketName(id);
+    }
+
+    this.bucket = new Bucket(scope, `${id}S3Bucket`, {
+      bucketName: this.bucketName,
+      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      encryption: BucketEncryption.KMS,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: true,
```

Comment:
are these properties all necessary to be hardcoded or should we expose some of them through `ProductStackAssetBucketProps`

Reply:
I think we need these configurations to make asset buckets work as we intend for our use case.  I think allowing customers to pass their own props might add further complication and could break the intended use case. There might be some customization that customers might want done on the asset bucket, but I don't think they are needed right now.